controllers/generic_service_handler.go

```diff
@@ -69,6 +72,25 @@ func (h *genericServiceHandler) createOrUpdate(a *aiven.Client, object client.Ob
 			UserConfig:            userConfig,
 		}
 
+		for _, s := range spec.ServiceIntegrations {
+			// Validates that read_replica is running
+			// If not, the wrapper controller will try later
+			if s.IntegrationType == "read_replica" {
```

Comment:
I reckon this IF statement should be moved to the `func (h *genericServiceHandler) checkPreconditions` and in this case we do not need additional error handling in the `controller.go`

Reply:
And also we have this helper function `checkServiceIsRunning`